schema/resource.go

```diff
@@ -117,16 +117,12 @@ func (resource *Resource) String() string {
 }
 
 //Update resource data
-func (resource *Resource) Update(updateData map[string]interface{}) error {
-	data := resource.properties
-	for _, property := range resource.schema.Properties {
-		id := property.ID
-
-		if val, ok := updateData[id]; ok {
-			data[id] = updateResourceRecursion(val, data[id])
-		}
+func (resource *Resource) Update(updateData map[string]interface{}) map[string]interface{} {
```

Comment:
This no longer updates the resource so maybe a new name is in order. Something like `MakeUpdatedCopy`, `CloneWithUpdate` etc.